benchmark/python/gluon/benchmark_gluon_rnn.py

```diff
@@ -0,0 +1,225 @@
+parser.add_argument('--latency', action='store_true', default=False,
+                    help='Measursing the latency, batchsize will be set to 1.')
+parser.add_argument('--train', action='store_true', default=False,
+                    help='Run backward benchmark.')
+parser.add_argument('--dropout', type=float, default=0, 
+                    help='float, default 0; Fraction of the input that gets dropped out during training time.')
+parser.add_argument('--gpu', action='store_true', default=False,
+                    help='bool, default Flase; set True to run benchmarks on GPU.')
+parser.add_argument('--no-hybridize', action='store_true', default=False,
+                    help='bool, default False; hybridize will introduce a better perf.')
+parser.add_argument('--bidirection', action='store_true', default=False,
+                    help='bool, default False; Whether to use bidirectional unroll.')
+
+opt = parser.parse_args()
+logging.basicConfig(level=logging.INFO)
+
+#[bs, sequence length, embedding size, hidden size]
+input_shape_list = [[64,15,500,500],
+   [64,20,500,500],
+   [64,25,500,500],
+   [64,30,500,500],
+   [64,35,500,500],
+   [64,40,500,500],
+   [64,45,500,500],
+   [64,50,500,500],
+   [16,25,512,512],
+   [32,25,512,512],
+   [64,25,512,512],
+   [128,25,512,512],
+   [16,25,1024,1024],
+   [32,25,1024,1024],
+   [64,25,1024,1024],
+   [128,25,1024,1024],
+   [16,25,2048,2048],
+   [32,25,2048,2048],
+   [64,25,2048,2048],
+   [128,25,2048,2048],
+   [16,25,4096,4096],
+   [32,25,4096,4096],
+   [64,25,4096,4096],
+   [128,25,4096,4096]]
+
+rnncell_type = ['lstm', 'gru', 'all']
+input_layout = ['TNC', 'NTC']
+
+if not opt.gpu:
+    ctx = mx.cpu()
+else:
+    ctx = mx.gpu(0)
+
+dropout = opt.dropout
+bidirection = opt.bidirection
+unfuse = opt.unfuse
+celltype = opt.cell_type
+
+dry_run = 20
+num_iter = 100
+
+def get_rnn_layer(input_shape, num_layer, cell_type, dropout=0, bidirection=False):
+    hidden_size = input_shape[3]
+    embedding_size = input_shape[2]
+    if cell_type == 'lstm':
+        rnn_layer = rnn.LSTM(hidden_size, num_layer, dropout=dropout,
+                             bidirectional=bidirection, input_size=embedding_size,
+                             prefix='_lstm_layer')
+    elif cell_type == 'gru':
+        rnn_layer = rnn.GRU(hidden_size, num_layer, dropout=dropout,
+                            bidirectional=bidirection, input_size=embedding_size,
+                            prefix='_gru_layer')
+    return rnn_layer
+
+
+def rnn_cell_score(input_shape, cell_type, ctx, num_layer, dropout=0, bidirection=False, layout='TNC', unfuse=False, hybridize=True, is_train=False):
+    bs = input_shape[0]
+    seq_len = input_shape[1]
+    embedding_size = input_shape[2]
+    hidden_size = input_shape[3]
+    rnn_layer = get_rnn_layer(input_shape, num_layer, cell_type, dropout, bidirection)
+    input_data = mx.sym.Variable('data')
+
+    if unfuse:
+        rnn_cell = rnn_layer._unfuse()
```

Comment:
the _unfuse() function is not intended to be exposed and we are thinking of ways to deprecate it. new features such as LSTM projection/state clipping are not supported in this function.